### PR TITLE
Make the Ballerina Project Tooling Friendly

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
## Examples
N/A
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/1746

### CODE REVIEW

File: build-config/resources/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+[[platform.java11.dependency]]
+path = "./lib/kafka_2.12-@kafka.version@.jar"
+groupId = "org.apache.kafka"
+artifactId = "kafka_2.12"
+version = "@kafka.version@"
+
+
+[[platform.java11.dependency]]
+path = "./lib/avro-1.9.2.jar"
```
Comment (ThisaruGuruge) [inline build-config/resources/Ballerina.toml:29]: I think we should handle the Avro version as same we handle the Kafka version, using the placeholders, and put the Avro version in the `gradle.properties` file.

File: ballerina/build.gradle [hunk 2]
```diff
@@ -82,20 +84,26 @@ dependencies {
...
 
-task revertTomlFiles {
+task commitTomlFiles {
     doLast {
-        ballerinaConfigFile.text = originalConfig
-        ballerinaCompilerPluginFile.text = originalCompilerPlugin
+        project.exec {
+            ignoreExitValue true
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                commandLine 'cmd', '/c', "git commit Ballerina.toml CompilerPlugin.toml -m '[Automated] Commit the native jar versions'"
```
Comment (ThisaruGuruge) [inline ballerina/build.gradle:102]:
```suggestion
                commandLine 'cmd', '/c', "git commit Ballerina.toml CompilerPlugin.toml -m '[Automated] Update the native jar versions'"
```

File: ballerina/build.gradle [hunk 3]
```diff
@@ -82,20 +84,26 @@ dependencies {
...
+            } else {
+                commandLine 'sh', '-c', "git commit Ballerina.toml CompilerPlugin.toml -m '[Automated] Commit the native jar versions'"
```
Comment (ThisaruGuruge) [inline ballerina/build.gradle:104]:
```suggestion
                commandLine 'sh', '-c', "git commit Ballerina.toml CompilerPlugin.toml -m '[Automated] Update the native jar versions'"
```

Comment (dilanSachi) [reply ballerina/build.gradle]: Changed in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/280/commits/2d0a36ee6ac39bd2f6eec146a23726b1eea39abe

File: ballerina/build.gradle (hunk 2 above)
Comment (dilanSachi) [reply ballerina/build.gradle]: Changed in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/280/commits/2d0a36ee6ac39bd2f6eec146a23726b1eea39abe

File: build-config/resources/Ballerina.toml (hunk 1 above)
Comment (dilanSachi) [reply build-config/resources/Ballerina.toml]: Changed in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/280/commits/2d0a36ee6ac39bd2f6eec146a23726b1eea39abe

File: build.gradle [hunk 4]
```diff
@@ -25,8 +25,8 @@ plugins {
 ext.ballerinaLangVersion = project.ballerinaLangVersion
 ext.kafkaVersion = project.kafkaVersion
 ext.awaitalityVersion = "3.1.6"
-ext.avroVersion = "1.9.2"
 ext.puppycrawlCheckstyleVersion = "8.18"
+ext.avroVersion = project.avroVersion
 ext.stdlibFileVersion = project.stdlibFileVersion
 ext.stdlibIoVersion = project.stdlibIoVersion
 ext.stdlibLogVersion = project.stdlibLogVersion
```
Comment (ThisaruGuruge) [inline build.gradle:32]: We can remove all these `ext.*` versions.

**Review by ThisaruGuruge — APPROVED** [review]

File: build.gradle (hunk 4 above)
Comment (dilanSachi) [reply build.gradle]: Removed in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/280/commits/a40cc516702bc802ad582fe4dc4d6c5f34ce06f4

File: ballerina/Ballerina.toml [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
+version = "3.0.0"
 distribution = "slbeta3"
 
 [[platform.java11.dependency]]
-path = "../native/build/libs/kafka-native-@project.version@.jar"
+path = "../native/build/libs/kafka-native-3.0.0-SNAPSHOT.jar"
 groupId = "org.ballerinalang.messaging.kafka"
 artifactId = "kafka"
-version = "@project.version@"
+version = "3.0.0-SNAPSHOT"
```
[the remark this answers is not in the document]
Comment (dilanSachi) [reply ballerina/Ballerina.toml]: Removed in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/280/commits/4f494eff6ad53ff4c8337c9d3ef8bbb43e4277f6

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**